server/mailet/integration-testing/src/test/java/org/apache/james/mailets/TemporaryJamesServer.java

```diff
@@ -61,6 +67,13 @@
         .putProcessor(CommonProcessors.error())
         .putProcessor(CommonProcessors.transport());
 
+    public static final Module WEBADMIN_IMAP_SMTP_MEMORY_SERVER = Modules.override(
+        MemoryJamesServerMain.SMTP_AND_IMAP_MODULE,
+        MemoryJamesServerMain.WEBADMIN)
+        .with(binder -> binder.bind(JwtConfiguration.class).toInstance(NO_JWT_CONFIGURATION),
```

Comment:
As far as I understand, the 3 overrides all belongs to webadmin.
The first two are related to authentication disabling.
The third setup the http server for testing (random port, IP, etc).
I would not create an barely reusable WEBADMIN_IMAP_SMTP_MEMORY_SERVER but a WEBADMIN_NO_AUTH_MODULE that only contains MemoryJamesServerMain.WEBADMIN and the 3 overrides.

Another much nicer solution would be to provide builders that actually make sense.